### DISCUSSION TITLE
API to fetch if ring base station is currently in alarm mode

### DISCUSSION THREAD

**Discussion by prashantuw** [dgreif/ring#763] [General]
Is there an API to know if the ring station is currently in alarm mode with siren on?

**Comment by dgreif** [comment]
It is possible if you are connected via WebSockets.  How you get to that point is complicated 😄.  Best option is to use a client like this one to get connected.

**Comment by prashantuw** [comment]
Thanks for info Dusty. Few clarifying questions:
1. Which client are you referring to? I am using ring-client-api already in my application.
2. Are there examples of how to use websockets for this usecase?
Really appreciate your help!

**Reply by dgreif** [reply]
Yep, referring to the `ring-client-api`.  You should be able to do something like `location.getDevices()`, then find devices of type `RingDeviceType.SecurityPanel`, and then look at the data for `alarmInfo` like I'm doing here: https://github.com/dgreif/ring/blob/master/homebridge/security-panel.ts#L109.  Sorry there isn't a more straightforward api for it at this point

**Reply by prashantuw** [reply]
Perfect, this is exactly what I was looking for. Thank you so much for these excellent pointers. Cheers!
